Mathlib/Analysis/Complex/UpperHalfPlane/Basic.lean

```diff
@@ -377,7 +380,14 @@ theorem denom_apply (g : SL(2, ℤ)) (z : ℍ) :
     denom g z = (↑g : Matrix (Fin 2) (Fin 2) ℤ) 1 0 * z + (↑g : Matrix (Fin 2) (Fin 2) ℤ) 1 1 := by
   simp [denom, coe']
 
-end ModularGroup
+@[simp]
+lemma denom_S (z : ℍ) : denom (ModularGroup.S) z = z.1 := by
```

Comment:
The `rfl` at the end of this proof suggests something is amiss; at that point the goal reads as `↑z = ↑z`, where the two arrows (Subtype.val and UpperHalfPlane.coe) are ultimately defeq, but are _not_ syntactically equal, despite printing as the same in the goal view!  I suggest changing the statement to `denom (ModularGroup.S) z = z` instead.